go/pkg/pwapi/api_team-accept-invite.go

```diff
@@ -4,13 +4,53 @@ import (
 	"context"
 
 	"pathwar.land/pathwar/v2/go/pkg/errcode"
+	"pathwar.land/pathwar/v2/go/pkg/pwdb"
 )
 
 func (svc *service) TeamAcceptInvite(ctx context.Context, in *TeamAcceptInvite_Input) (*TeamAcceptInvite_Output, error) {
-	if in == nil || in.TeamMemberID == 0 {
+	if in == nil || in.TeamInviteID == "" {
 		return nil, errcode.ErrMissingInput
 	}
 
+	userID, err := userIDFromContext(ctx, svc.db)
+	if err != nil {
+		return nil, errcode.ErrUnauthenticated.Wrap(err)
+	}
+
+	teamInviteID, err := pwdb.GetIDBySlugAndKind(svc.db, in.TeamInviteID, "team-invite")
+	if err != nil {
+		return nil, errcode.TODO.Wrap(err)
+	}
+
+	// check that invite exists
+	var teamInvite pwdb.TeamInvite
+	err = svc.db.
+		Where(&pwdb.TeamInvite{
+			ID:     teamInviteID,
+			UserID: userID,
+		}).
+		First(&teamInvite).
+		Error
+	if err != nil {
+		return nil, errcode.ErrTeamInviteDoesNotExist.Wrap(err)
+	}
+
+	// create team member
+	err = svc.db.
+		Create(&pwdb.TeamMember{
+			UserID: userID,
+			TeamID: teamInvite.TeamID,
+		}).Error
+	if err != nil {
+		return nil, errcode.ErrCreateTeamMember.Wrap(err)
+	}
+
+	// remove invite
```

Comment:
you should update a state instead of deleting here

if you don't want to use a status and prefer to delete, then you need to also delete other team invites on the same season